includes/Order/Hooks.php

```diff
@@ -292,7 +292,9 @@ public function ensure_vendor_coupon( $valid, $coupon ) {
         foreach ( WC()->cart->get_cart() as $item ) {
             $product_id = $item['data']->get_id();
 
-            $available_vendors[]  = (int) get_post_field( 'post_author', $product_id );
+            $vendor = dokan_get_vendor_by_product( $product_id );
```

Comment:
@davefx 
`$available_vendors[]  = dokan_get_vendor_by_product( $product_id, true );` is enough